src/nvim/lib/kvec.h

```diff
@@ -58,7 +58,7 @@
     }
 
 #define kv_init(v) ((v).size = (v).capacity = 0, (v).items = 0)
-#define kv_destroy(v) xfree((v).items)
+#define kv_destroy(v) XFREE_CLEAR((v).items)
```

Comment:
it seems weird to only reset the pointer and not the other fields: the data structure is now in an inconsistent state. Maybe `(xfree((v).items), kv_init(v))`.

Reply:
c4f56c4